README.md

```diff
@@ -1,16 +1,71 @@
 # Flat File Creator
 
-## It allows to generate a flat positional file, using an object that defines its structure
+This library allows to generate a flat positional file, using an array that defines it's structure
 
-`getAsyncFlatFileCreator(maps: [MapDefinition], options: FlatFileOptions)` :
-FlatFileCreator
+### Generate your asynchronous file creator:
+The method `getAsyncFlatFileCreator` returns a function configured through the two required parameters, maps and options. The maps array parameter will contain the definition of the line structure of the text file, options allows you to configure some general options of the file that will be created.
 
-- maps: positional flat file definition
-- options: additional options (optional)
+`getAsyncFlatFileCreator(maps: [MapDefinitioObject], options: FlatFileOptions)`: FlatFileCreator
 
-`FlatFileCreator(data: [any], path: String)`
+#### `"maps": [MapDefinitioObject]` - positional row definition param 
+The array is made up of objects defines the composition of each line.
+
+The necessary attributes of the objects contained in the array are:
+
+- `name: string` this attribute is the reference to the name of the attribute that must be present in the dataset that will be passed to the generated function to process the value and position it in the desired point;
+- `size: numeric` is the total dimension that the field will have in the generated file;
+- `type: string ` is the reference to the data type related to the field and can have the following values: 
+
+  - `string`
+  - `float`
+  - `integer`
+  - `date`
+  
+  the other attributes that are needed depend on the type chosen
+  
+#### `options: object` additional options (optional)
+Option object is optional if it not will provided file will be generated with default option.
+
+- rowEnd: defines the terminator character of each line; `default: ''`
+- encoding: it's relative to the file encoding provided by the fs node module;`default: 'utf8'`
+- mode: it's relative to the file save mode provided by the fs node module;`default: 0o666`
+- flag: it's relative to the file save flag provided by the fs node module;`default: 'a'`
+  
+#### Data types
+specific attribute required by chosen data type
+
+#### `string`
```

Comment:
I'm sorry maybe the way I put things off-road. I wanted first to conclude the general explanation of the parameters for the `getAsyncFlatFileCreator` function.
The attributes specified in the `Data types` section must be inserted in each mapping objects together with the type choice, for each field you add, because you may have fields that have different requests for the same data type, or you can leave as default. Maybe it is more clear if I will move this section under maps.